lib/utils.js

```diff
@@ -5,6 +5,48 @@ module.exports = {
         return crypto.createHash("md5").update(str, "utf8").digest("hex")
     },
 
+    checkUniqueHash: function(items, source, properties) {
+        for (let item of items) {
+            // This will save some time...
+            if (item._id === source._id) {
+                let itemHash = []
+                let sourceHash = []
+
+                // Add one to the properties length to add an additional static check.
+                for (var i = 0; i < (properties.length + 1); i++) {
```

Comment:
minor nitpick, but I'd change **var** to **let** because I don't see a reason why you'd want to hoist **i**. This can introduce some unexpected/unintended behavior.

If you know how variable hoisting works in JS you can ignore this, if not this saves one Google search :).

**var** declares the variable in the entire function not just the scope:

```js
function f() {
    for(let i = 0; i < 10; i++) {
        for(var j = 0; j < 10; j++) {
            // do shit
        }
    }
    
    console.log(j); // works fine even though we're not in the scope anymore, thank you var? :P
    console.log(i); // this will throw an ReferenceError because i was used with let which doesn't hoist the variable
}

f();

// j doesn't exist here obviously...
```

Reply:
I'll change the code template I use. Simply a side effect of using the "for" template. Will fix.